File: Leetcode/AddBinary.cs
// Given two binary strings a and b, return their sum as a binary string.

public class Solution 
{
    public string AddBinary(string a, string b) 
    {
        var binary = new StringBuilder();
        int carry = 0;
        int i = a.Length - 1, j = b.Length - 1;
        
        while (i >= 0 || j >= 0)
        {
            int sum = carry;
            sum += i >= 0 ? a[i] - '0' : 0;
            sum += j >= 0 ? b[j] - '0' : 0;
            binary.Insert(0, (sum % 2).ToString());
            carry = sum / 2;
            
            i--;
            j--;
        }
        
        if (carry != 0) binary.Insert(0, carry.ToString());
        return binary.ToString();
    }
}

File: Leetcode/AddBinary.py
# Given two binary strings a and b, return their sum as a binary string.

class Solution(object):
    def addBinary(self, a, b):
        binary, carry = "", 0
        i, j = len(a) - 1, len(b) - 1
        
        while (i >= 0 or j >= 0):
            sum = carry
            sum += int(a[i]) if i >= 0 else 0
            sum += int(b[j]) if j >= 0 else 0
            binary += str(sum % 2)
            carry = sum // 2
            
            i -= 1
            j -= 1
        
        if carry != 0:
            binary += str(carry)
        return binary[::-1]